src/components/organisms/modals/UserAndRoleManagementModal/UserAndRoleManagementModal.tsx

```diff
@@ -42,15 +44,22 @@ const UserAndRoleManagementModal: FC<UserAndRoleManagementModalProps> = ({
           appearance: AppearanceTypes.Secondary,
           children: cancelButtonContent || t('button.no'),
           size: SizeTypes.M,
-          onClick: closeModal,
+          onClick: () => {
+            if (handleCancel) {
+              handleCancel()
+            } else {
+              closeModal()
+            }
+          },
         },
         {
           appearance: AppearanceTypes.Primary,
           onClick: () => {
             if (handleProceed) {
               handleProceed()
+            } else {
+              closeModal()
```

Comment:
Currently when the user is beeing archived, the modal doesn't close.

Reply:
Yeah skipped this for now, since I noticed there are a lot of changes coming to the archive part from your branch. But if we merge your branch first, then I'll add it